demo/index.html

```diff
@@ -40,7 +40,9 @@
           this._openDialog1 = this._openDialog1.bind(this);
           this._openDialog2 = this._openDialog2.bind(this);
           this._openDialog3 = this._openDialog3.bind(this);
+          this._openDialog4 = this._openDialog4.bind(this);
```

Comment:
I do think it would be better to give the dialogs names instead of just iterating. :)